resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/kubernetes/KubernetesClusterSchedulerBackend.scala

```diff
@@ -175,69 +175,61 @@ private[spark] class KubernetesClusterSchedulerBackend(
     val executorCpuQuantity = new QuantityBuilder(false)
       .withAmount(executorCores)
       .build()
-    val requiredEnv = new ArrayBuffer[EnvVar]
-    requiredEnv += new EnvVarBuilder()
-      .withName("SPARK_EXECUTOR_PORT")
-      .withValue(executorPort.toString)
-      .build()
-    requiredEnv += new EnvVarBuilder()
-      .withName("SPARK_DRIVER_URL")
-      .withValue(driverUrl)
-      .build()
-    requiredEnv += new EnvVarBuilder()
-      .withName("SPARK_EXECUTOR_CORES")
-      .withValue(executorCores)
-      .build()
-    requiredEnv += new EnvVarBuilder()
-      .withName("SPARK_EXECUTOR_MEMORY")
-      .withValue(executorMemory)
-      .build()
-    requiredEnv += new EnvVarBuilder()
-      .withName("SPARK_APPLICATION_ID")
-      .withValue(applicationId())
-      .build()
-    requiredEnv += new EnvVarBuilder()
-      .withName("SPARK_EXECUTOR_ID")
-      .withValue(executorId)
-      .build()
-    val requiredPorts = new ArrayBuffer[ContainerPort]
-    requiredPorts += new ContainerPortBuilder()
-      .withName(EXECUTOR_PORT_NAME)
-      .withContainerPort(executorPort)
-      .build()
-    requiredPorts += new ContainerPortBuilder()
-      .withName(BLOCK_MANAGER_PORT_NAME)
-      .withContainerPort(blockmanagerPort)
-      .build()
-    (executorKubernetesId, kubernetesClient.pods().createNew()
-      .withNewMetadata()
-        .withName(name)
-        .withLabels(selectors)
-        .withOwnerReferences()
-        .addNewOwnerReference()
-          .withController(true)
-          .withApiVersion(driverPod.getApiVersion)
-          .withKind(driverPod.getKind)
-          .withName(driverPod.getMetadata.getName)
-          .withUid(driverPod.getMetadata.getUid)
-        .endOwnerReference()
-      .endMetadata()
-      .withNewSpec()
-        .addNewContainer()
-          .withName(s"exec-${applicationId()}-container")
-          .withImage(executorDockerImage)
-          .withImagePullPolicy("IfNotPresent")
-          .withNewResources()
-            .addToRequests("memory", executorMemoryQuantity)
-            .addToLimits("memory", executorMemoryLimitQuantity)
-            .addToRequests("cpu", executorCpuQuantity)
-            .addToLimits("cpu", executorCpuQuantity)
-            .endResources()
-          .withEnv(requiredEnv.asJava)
-          .withPorts(requiredPorts.asJava)
-          .endContainer()
-        .endSpec()
-      .done())
+    val requiredEnv = Seq(
```

Comment:
There's probably some helper methods to pull out here too - creating a `KubernetesUtils` class seems potentially useful.